loopout.go

```diff
@@ -719,6 +733,45 @@ func (s *loopOutSwap) waitForHtlcSpendConfirmed(globalCtx context.Context,
 		case err := <-spendErr:
 			return nil, err
 
+		// Receive status updates for our payment so that we can detect
+		// whether we've successfully pushed our preimage.
+		case status, ok := <-trackChan:
+			// If our channel has been closed, indicating that the
+			// server is finished providing updates because the
+			// payment has reached a terminal state, we replace
+			// the closed channel with nil so that we will no longer
+			// listen on it.
+			if !ok {
+				trackChan = nil
```

Comment:
Do we need to check here that we actually reached a terminal state?

Reply:
I don't think so, grpc only sends an EOF when the stream exits cleanly. Checking state here is checking for lnd trackpayments bugs in loop, I think it's a safe assumption to make.